coral-trino/src/test/java/com/linkedin/coral/trino/trino2rel/TrinoToRelConverterTest.java

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright 2021-2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.trino2rel;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.plan.RelOptUtil;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.linkedin.coral.hive.hive2rel.functions.StaticHiveFunctionRegistry;
+
+import static com.linkedin.coral.trino.trino2rel.ToRelTestUtils.*;
+import static com.linkedin.coral.trino.trino2rel.Trino2CoralOperatorTransformerMapUtils.*;
+import static org.apache.calcite.sql.type.OperandTypes.*;
+import static org.testng.AssertJUnit.assertEquals;
+
+
+public class TrinoToRelConverterTest {
+  private static HiveConf conf;
+
+  @BeforeClass
+  public void beforeClass() throws HiveException, IOException, MetaException {
+    // Simulating a Coral environment where "foo" exists
+    StaticHiveFunctionRegistry.createAddUserDefinedFunction("foo", ReturnTypes.INTEGER,
+        or(NILADIC, NUMERIC, NUMERIC_NUMERIC));
+
+    conf = ToRelTestUtils.loadResourceHiveConf();
+    ToRelTestUtils.initializeViews(conf);
+
+    Map<String, OperatorTransformer> TRANSFORMER_MAP = Trino2CoralOperatorTransformerMap.TRANSFORMER_MAP;
+
+    // foo(a) or foo()
+    createTransformerMapEntry(TRANSFORMER_MAP, createOperator("foo", ReturnTypes.INTEGER, or(NILADIC, NUMERIC)), 1,
+        "foo", null, null);
+
+    // foo(a, b) => foo((10 * a) + (10 * b))
+    createTransformerMapEntry(TRANSFORMER_MAP, createOperator("foo", ReturnTypes.INTEGER, NUMERIC_NUMERIC), 2, "foo",
+        "[{\"op\":\"+\",\"operands\":[{\"op\":\"*\",\"operands\":[{\"value\":10},{\"input\":1}]},{\"op\":\"*\",\"operands\":[{\"value\":10},{\"input\":2}]}]}]",
+        null);
+  }
+
+  @AfterTest
+  public void afterClass() throws IOException {
+    FileUtils.deleteDirectory(new File(conf.get(CORAL_FROM_TRINO_TEST_DIR)));
+  }
+
+  @DataProvider(name = "support")
+  public Iterator<Object[]> getSupportedSql() {
```

Comment:
I have not checked the correctness of the output (expected result). Have you manually inspected that it looks good?

Reply:
I have checked for some correctness (i.e are all the columns we expect to be there, there?) but it definitely hasn't been rigorous. The problem is I can't find good Calcite documentation/examples so I can check for correctness rigorously. How was this done in HiveToRel?

Reply:
One idea is to convert back to SQL and read the query if it looks correct. The flip side of the transformation (CoralIR to Trino SQL) should have a transformation from "TrinoRel" (not "CoralRel") to Trino SQL ("TrinoRel" is the transformed version of "CoralRel" after applying the (coral-to-trino) transformation map). You could provide your Rel to that part and see how the SQL looks like.